Forge fixes

matrei opened the pull request:
Resolves #14972

matrei committed:
fix(deps): `micronaut` v. `4.6.5` -> `4.9.10`

matrei committed:
fix(deps): `micronaut-serde-jackson` v. `2.11.0` -> `2.15.1`

matrei committed:
chore: group and sort `gradle.properties`

matrei committed:
fix(deps): `slf4j` v. `2.0.16` -> `2.0.17`

matrei committed:
chore: sort `gradle.properties`

matrei committed:
fix(forge): adjust shadow plugin coords

matrei committed:
fix(forge): adjust feature categories

matrei committed:
chore: typo

jdaugherty commented on grails-forge/grails-forge-core/src/main/resources/pom.xml:
```diff
@@ -55,8 +55,8 @@
+            <version>9.0.2</version>
         </dependency>
         <dependency>
             <groupId>com.github.erdi</groupId>
```
Looking over this view - erdi caught my eye. I didn't think we use this anymore since it's incompatible, should the web driver binaries plugin be removed?

matrei replied:
Yes, I think you are right as `webdriver-binaries-gradle-plugin` is not working with Gradle 8.7+ and the project seems to be dead: erdi/webdriver-binaries-gradle-plugin#44
I'll remove it.

matrei replied:
@jdaugherty Actually, I see that @jamesfredley has put in a lot of effort to keep the `webdriver-binaries` solution alive a bit longer so I reverted that removal and let him decide whether it stays or goes.

jamesfredley replied:
I think we should keep webdriver-binaries as a non-default, optional feature for Grails 7 and then remove it for Grails 8.  That gives users transition time and it works with up to Gradle 8.6 with up to JDK 21.

Language in the title and description about these limitations.

https://github.com/apache/grails-core/blob/00b8008c907807502f2d5a7605f5ae7aa3745907/grails-forge/grails-forge-core/src/main/java/org/grails/forge/feature/test/GebWithWebDriverBinaries.java#L53-L62

jamesfredley replied:
There is also code which adjusts the Gradle and Java version when `geb-with-webdriver-binaries` is selected, with comments in generated config.

jdaugherty replied:
Makes sense.  Thank you!

matrei committed:
fix(forge): remove Geb with Webdriver Binaries

`webdriver-binaries-gradle-plugin` is not compatible with Gradle 8.7+
and there does not seem to be any plans to continue the project.

matrei committed:
Revert "fix(forge): remove Geb with Webdriver Binaries"

This reverts commit bdc56816dec83d518908cfc825d544933c80a4f9.

jamesfredley commented on grails-forge/grails-forge-core/src/main/java/org/grails/forge/feature/other/ShadePlugin.java:
```diff
@@ -68,7 +68,7 @@ public void apply(GeneratorContext generatorContext) {
         GradlePlugin.Builder builder = GradlePlugin.builder()
                 .id("com.gradleup.shadow")
-                .lookupArtifactId("shadow");
+                .lookupArtifactId("com.gradleup.shadow.gradle.plugin");
```
This looks slightly different on the other plugins referenced by id, with a version from POM.xm.

https://github.com/apache/grails-core/blob/00b8008c907807502f2d5a7605f5ae7aa3745907/grails-forge/grails-forge-core/src/main/java/org/grails/forge/feature/reloading/Jrebel.java#L47-L50

https://github.com/apache/grails-core/blob/00b8008c907807502f2d5a7605f5ae7aa3745907/grails-forge/grails-forge-core/src/main/java/org/grails/forge/feature/asciidoctor/Asciidoctor.java#L65-L68

matrei replied:
Yes, you are right, I used the wrong one: https://mvnrepository.com/artifact/com.gradleup.shadow
It does not matter in this case but I'll change it to `shadow-gradle-plugin`.

matrei committed:
fix: correct artifactId for `com.gradleup.shadow`

matrei committed:
fix(deps): `asciidoctor-gradle-jvm` v. `4.0.4` -> `4.0.5`
